### PR TITLE
fix(kit): add typings to kit server middleware

### PR DESCRIPTION
**Pull request by Rigo-m**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add types to `addServerMiddleware`.
I've discovered that I can place an object with `path` and `handler` inside `addServerMiddleware` by taking a look at tailwindcss nuxt module. I think this should be auto-documented via types.

I need help to better type this, I'm not so ts-savvy yet.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/kit/src/server.ts [hunk 1]
```diff
@@ -1,6 +1,12 @@
 import { useNuxt } from './context'
+import { IncomingMessage, ServerResponse } from 'http'
+
+export interface ServerMiddleware {
+  path: string,
+  handler: (req: IncomingMessage, res: ServerResponse) => void
```
Comment (pi0) [inline packages/kit/src/server.ts:6]: Handler can be also be a string (path to middleware). Reference types: https://github.com/nuxt/framework/blob/e2e021cae68b1e62f0d7d8ecfb9ca6b4889ff9d5/packages/nitro/src/server/middleware.ts#L9

Comment (Rigo-m) [reply packages/kit/src/server.ts]: I have improved the handler type. Although, I wonder, why not importing directly the ServerMiddleware interface from the nitro package?

Comment (pi0) [reply packages/kit/src/server.ts]: Thanks. Nitro package will be dropped soon but indeed later we could depend on the types from nitropack. Do you mind to also use middleware type from h3?

Comment (Rigo-m) [reply packages/kit/src/server.ts]: Check the latest commit

**Pull request merged**